common/lib/plugin_service.ts

```diff
@@ -114,17 +117,29 @@ export class PluginService implements ErrorHandler, HostListProviderService {
       this._currentHostInfo = this._initialConnectionHostInfo;
 
       if (!this._currentHostInfo) {
-        if (this.getHosts().length === 0) {
+        if (this.getAllHosts().length === 0) {
           throw new AwsWrapperError(Messages.get("PluginService.hostListEmpty"));
         }
 
-        const writerHost = getWriter(this.getHosts());
+        const writerHost = getWriter(this.getAllHosts());
+        if (!this.getHosts().some((hostInfo: HostInfo) => hostInfo.host === writerHost?.host)) {
+          throw new AwsWrapperError(
+            Messages.get(
+              "PluginService.currentHostNotAllowed",
+              this._currentHostInfo ? this._currentHostInfo.host : "<null>",
```

Comment:
Should this be `writerHost.host` instead of `currentHostInfo.host`? The new writer is the one that isn't allowed.